src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -241,6 +240,13 @@ public void storeContext(Context context, HttpServletRequest request) {
 			M_log.info(String.format("The user \"%s\" is account admin in the course %s",
 					request.getParameter(Utils.LTI_PARAM_UNIQNAME), request.getParameter(Utils.LTI_PARAM_CANVAS_COURSE_ID)));
 		}
+		ToolAccessGranter accessGranter = new ToolAccessGranter(request.getParameter(Utils.LTI_PARAM_CANVAS_COURSE_ID),
+				request.getParameter(Utils.LTI_PARAM_CANVAS_USER_ID));
+		boolean allowedToolAccess=accessGranter.isAllowedAccess(admin);
+		if (allowedToolAccess) {
```

Comment:
It's not clear what happens if they don't have access.   Are they allowed to keep going? Or should this set the value of Utils.IS_TOOL_ACCESS_ALLOWED and just return / go to error page ?  The consequences of not being allowed should get a note here.

Reply:
The plan was to pass the variable into the Velocity context, omit from the response any information other than a message saying that the current user does not have permission to use the tool. See: 

https://github.com/tl-its-umich-edu/canvas_course_manager/pull/218

Reply:
+1